src/drivers/fluid_sdl3.c

```diff
@@ -0,0 +1,449 @@
+    }
+    else if(fluid_settings_str_equal(settings, "audio.sample-format", "16bits"))
+    {
+        FLUID_LOG(FLUID_DBG, "Selected 16 bit sample format");
+
+        sample_size = sizeof(short);
+        write_ptr   = fluid_synth_write_s16;
+
+        aspec.format = SDL_AUDIO_S16;
+    }
+    else
+    {
+        FLUID_LOG(FLUID_ERR, "Unhandled sample format");
+        return NULL;
+    }
+
+    /* Compile the format buffer */
+    aspec.channels   = 2;
+    /*aspec.samples    = aspec.channels * ((period_size + 7) & ~7);
+    aspec.callback   = (SDL_AudioCallback)SDLAudioCallback;*/
+
+    /* Set default device to use */
+    device   = NULL;
+    dev_name = NULL;
+
+    /* create and clear the driver data */
+    dev = FLUID_NEW(fluid_sdl3_audio_driver_t);
+
+    if(dev == NULL)
+    {
+        FLUID_LOG(FLUID_ERR, "Out of memory");
+        return NULL;
+    }
+
+    FLUID_MEMSET(dev, 0, sizeof(fluid_sdl3_audio_driver_t));
+
+    /* get the selected device name. if none is specified, use default device. */
+    if(fluid_settings_dupstr(settings, "audio.sdl3.device", &device) == FLUID_OK
+            && device != NULL && device[0] != '\0')
+    {
+        int n, nDevs = GetNumAudioDevices(0, dev);
+
+        for(n = 0; n < nDevs; n++)
+        {
+            dev_name = SDLC_GetAudioDeviceName(n, 0, dev);
+
+            if(FLUID_STRCASECMP(dev_name, device) == 0)
+            {
+                FLUID_LOG(FLUID_DBG, "Selected audio device GUID: %s", dev_name);
+                break;
+            }
+        }
+
+        if(n >= nDevs)
+        {
+            FLUID_LOG(FLUID_DBG, "Audio device %s, using \"default\"", device);
+            dev_name = NULL;
+        }
+    }
+
+    if(device != NULL)
+    {
+        FLUID_FREE(device);
+    }
+
+    do
+    {
+        /* Save copy of synth */
+        dev->synth = synth;
+
+        /* Save copy of other variables */
+        dev->write_ptr = write_ptr;
+        dev->frame_size = sample_size * aspec.channels;
+
+        /* Open audio device */
+        dev->stream = SDL_OpenAudioDeviceStream(SDL_AUDIO_DEVICE_DEFAULT_PLAYBACK, &aspec, SDLAudioCallback, NULL);
+        dev->devid = SDL_GetAudioStreamDevice(dev->stream);
+
+        if(!dev->stream)
+        {
+            FLUID_LOG(FLUID_ERR, "Failed to open audio stream");
+            break;
+        }
+
+        /* Start to play */
+        SDL_ResumeAudioStreamDevice(dev->stream);
+
+        return (fluid_audio_driver_t *) dev;
+    } while(0);
+
+    delete_fluid_sdl3_audio_driver(&dev->driver);
+    return NULL;
+}
+
+
+void delete_fluid_sdl3_audio_driver(fluid_audio_driver_t *d)
+{
+    fluid_sdl3_audio_driver_t *dev = (fluid_sdl3_audio_driver_t *) d;
+
+    if(dev != NULL)
+    {
+        if(dev->devid)
+        {
+            /* Stop audio and close */
+            SDL_PauseAudioDevice(dev->devid);
+            SDL_CloseAudioDevice(dev->devid);
+        }
+
+        FLUID_FREE(dev);
+    }
+
+    SDL_QuitSubSystem(SDL_INIT_AUDIO);
```

Comment:
```suggestion
        SDL_QuitSubSystem(SDL_INIT_AUDIO);
    }
```

A malicious user or app could have triggered calls to QuitSubSystem by just passing NULL pointers to this function, forcing the ref count to zero and therefore eventually crashing SDL while audio is still in use. Guard it by NULL check to avoid that.